src/pages/StocktakeEditPage.js

```diff
@@ -115,12 +116,22 @@ export class StocktakeEditPage extends React.Component {
     switch (key) {
       default:
         return stocktakeItem[key];
-      case 'countedTotalQuantity':
+      case 'countedTotalQuantity': {
+        let cellContents;
+        if (!stocktakeItem.hasCountedBatches) {
+          // If not editable (i.e. finalised) we want to show 0 rather than '' as
+          // placeholder isn't shown
+          cellContents = isEditable ? '' : 0;
```

Comment:
I am pretty sure if stocktakeItem does not have counted batches it gets removed during finalisation.

Reply:
Nah the item remains annoyingly. If you expand it there are simply no rows. Maybe that's the bug I should be fixing!

Reply:
I've tried numerous times (on master), If item has no counted quantity it get's removed from finalised stock take.  Can you please give a use case of where it remains please (on master).

Reply:
So here are some items getting stocktake.
![image](https://user-images.githubusercontent.com/7684221/37808533-b2664408-2eaf-11e8-89ef-55bd42f3ced9.png)
Result:
![image](https://user-images.githubusercontent.com/7684221/37808545-cbd163be-2eaf-11e8-9aac-563c75fc0f63.png)

So if the snapshot was 0, then it'll be included with no stocktakeItemBatches. If snapshot was more than 0 it works fine

Reply:
So true. Nicely spotted

Reply:
I don't think it's a bug though, i have a distant recollection we wanted to keep them ?

Reply:
@andreievg I just spotted while sifting through code

(stocktake.js:adjustInventory)
```
// Delete all 'fresh' StocktakeBatches that have been created by stocktake
// but have not been changed
database.delete('StocktakeBatch', stocktakeBatches.filter(stocktakeBatch =>
  stocktakeBatch.snapshotTotalQuantity === 0 && stocktakeBatch.difference === 0));
```
So yea, it does check but doesn't touch the stocktakeItems, just clears out the batches.